driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -34,19 +35,26 @@
     private final ReadBinding binding;
     private final ChangeStreamOperation<T> changeStreamOperation;
 
-    private BsonDocument resumeToken;
+    private BsonTimestamp initialStartAtOperationTime;
     private BatchCursor<RawBsonDocument> wrapped;
+    private BsonDocument resumeToken;
 
     ChangeStreamBatchCursor(final ChangeStreamOperation<T> changeStreamOperation,
                             final BatchCursor<RawBsonDocument> wrapped,
                             final ReadBinding binding) {
         this.changeStreamOperation = changeStreamOperation;
-        this.resumeToken = changeStreamOperation.getResumeToken();
+        this.initialStartAtOperationTime = changeStreamOperation.getStartAtOperationTime();
         if (changeStreamOperation.getStartAfter() != null) {
-            this.resumeToken = changeStreamOperation.getStartAfter();
+            resumeToken = changeStreamOperation.getStartAfter();
+            changeStreamOperation.startAfter(null);
```

Comment:
We do this right before resuming.  No need to duplicate it here, right?

Reply:
We have to cache the resume token here if `startAfter` or `resumeAfter` were set. Otherwise, we would not have a resume token when resuming. If we do not cache the resume token here, then `startAfter` and `resumeAfter` will be cleared before resuming.

Reply:
But we don't need to set changeStreamOperation.startAfter to null here, do we?

Reply:
That is true. I thought you were referring to the entire code block. I will remove those.

Reply:
Removed.